### PR TITLE
ROS2 Migration

### LINKED ISSUE #76: any ROS2 port?

### PR DESCRIPTION
**Pull request by wep21**
Closes #76

### CODE REVIEW

**Comment by wep21** [comment]
@wxmerkt Could you create `ros2` branch? I will change the target branch to `ros2`.

**Comment by wep21** [comment]
@wxmerkt This PR is now ready to be reviewed. I appreciate it if you could review this.

**Comment by wep21** [comment]
CI result on my fork
https://github.com/wep21/octomap_mapping/actions/runs/1591345804

**Comment by wxmerkt** [comment]
@wep21 Thank you very much for the contribution. I'd like to test this locally, likely this weekend or next week. I'll circle back to this PR then. I hope this is okay

File: octomap_server/src/octomap_server.cpp [hunk 1]
```diff
@@ -309,7 +309,9 @@ OctomapServer::OctomapServer(const rclcpp::NodeOptions & node_options)
     std::make_shared<tf2_ros::TransformListener>(*tf2_buffer_);
 
   using std::chrono_literals::operator""s;
-  point_cloud_sub_.subscribe(this, "cloud_in");
+  rclcpp::SensorDataQoS qos_subscription;
+  // qos_subscription.keep_last(input_queue_size_);
+  point_cloud_sub_.subscribe(this, "cloud_in", qos_subscription.get_rmw_qos_profile());
```
Comment (wep21) [inline octomap_server/src/octomap_server.cpp]: @wxmerkt You can use `rmw_qos_profile_sensor_data` directly if you won't change the queue size.
https://github.com/ros2/rmw/blob/9f3c3e1b010a95013eb119ce2ae9b225bdfdc1ba/rmw/include/rmw/qos_profiles.h#L25

Comment (wxmerkt) [reply octomap_server/src/octomap_server.cpp]: Thank you for the hint! I think the PR is now good to go if you are happy with my changes.

**Review by wxmerkt — APPROVED** [review]
Thank you very much for your contribution and this effort in migrating to ROS2 @wep21. If you are happy with the changes I've added we can merge and release it.

**Comment by wep21** [comment]
@wxmerkt Thanks, please go ahead.

**Pull request merged**